### PR TITLE
go/staking: Add DebondingStartEscrowEvent

### PR DESCRIPTION
**Pull request by kostko**
This makes it easier to reason about the debonded amounts, both in shares and
in base units.

### CODE REVIEW

File: go/consensus/tendermint/apps/staking/transactions.go [hunk 1]
```diff
@@ -404,6 +405,15 @@ func (app *stakingApplication) reclaimEscrow(ctx *api.Context, state *stakingSta
 		}
 	}
```
Comment (tjanez) [inline go/consensus/tendermint/apps/staking/transactions.go:406]: Perhaps also add logging (like for `AddEscrow`) along these lines:

```go
	ctx.Logger().Debug("ReclaimEscrow: started debonding stake",
		"to", toAddr,
		"from", reclaim.Account,
		"base_units", stakeAmount,
		"active_shares", reclaim.Shares,
		"debonding_shares", debondingShares,
	)
```

Comment (kostko) [reply go/consensus/tendermint/apps/staking/transactions.go:406]: Done.

File: go/consensus/tendermint/apps/staking/api.go [hunk 2]
```diff
@@ -41,6 +41,9 @@ var (
 	// (value is an api.AddEscrowEvent).
 	KeyAddEscrow = stakingState.KeyAddEscrow
 
+	// KeyDebondingStart is an ABCI event attribute key for DebondingStartEvents.
```
Comment (tjanez) [inline go/consensus/tendermint/apps/staking/api.go:44]:
```suggestion
	// KeyDebondingStart is an ABCI event attribute key for DebondingStartEscrowEvents.
```

File: go/consensus/tendermint/staking/staking.go [hunk 3]
```diff
@@ -369,6 +369,16 @@ func EventsFromTendermint(
 
 				evt := &api.Event{Height: height, TxHash: txHash, Escrow: &api.EscrowEvent{Add: &e}}
 				events = append(events, evt)
+			case bytes.Equal(key, app.KeyDebondingStart):
+				// Debonding start event.
```
Comment (tjanez) [inline go/consensus/tendermint/staking/staking.go:373]:
```suggestion
				// Debonding start escrow event.
```

File: go/consensus/tendermint/staking/staking.go [hunk 4]
```diff
@@ -369,6 +369,16 @@ func EventsFromTendermint(
...
+				var e api.DebondingStartEscrowEvent
+				if err := cbor.Unmarshal(val, &e); err != nil {
+					errs = multierror.Append(errs, fmt.Errorf("staking: corrupt DebondingStart event: %w", err))
```
Comment (tjanez) [inline go/consensus/tendermint/staking/staking.go:376]:
```suggestion
					errs = multierror.Append(errs, fmt.Errorf("staking: corrupt DebondingStart escrow event: %w", err))
```

**Review by tjanez — APPROVED** [review]

**Review by Yawning — APPROVED** [review]
Seems fine to me.

**Pull request merged**